terraform/eks_arm64_cluster_setup/main.tf

```diff
@@ -0,0 +1,114 @@
+# ------------------------------------------------------------------------
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+terraform {
+  required_providers {
+    kubectl = {
+      source  = "gavinbunney/kubectl"
+      version = ">= 1.7.0"
+    }
+  }
+}
+
+module "common" {
+  source = "../common"
+}
+
+provider "aws" {
+  region = var.region
+}
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = "${var.eks_cluster_name}-vpc"
+  cidr = "10.0.0.0/16"
+
+  azs             = ["${var.region}a", "${var.region}b", "${var.region}c"]
+  private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets  = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+
+  enable_nat_gateway = true
+  enable_vpn_gateway = true
+  single_nat_gateway = true
+}
+
+module "eks" {
+  source = "terraform-aws-modules/eks/aws"
+
```

Comment:
We need to pin a version hwere for the eks module. We ran into the same issue with the `adot-op-cluster` where breaking changes made us unable to deploy. @sethAmazon  do you happen to have the version available? Would it be the same as the `adot-op-cluster` because we were dealing with the issue around the same time this PR was filed.